### DISCUSSION TITLE
Json RPC

### DISCUSSION THREAD

**Discussion by aKardasz** [fastapi/fastapi#5364] [Questions]
Having played around with [TRPC](https://trpc.io/ ), I feel like it would allow for a change in the way api's are written, especially for frontend. Thinking about it though, and running into my own problems trying to make an api trying to follow best practices, sometimes it seems like a rigid rest style is not always the best way to handle things.

So what the idea is to implement something similar. However it could utilize https://www.jsonrpc.org/specification. An example of this is 
https://github.com/pavlov99/json-rpc, since decorators are already used, I was curious to what the difficulty would be to add something like @jrpc_method or something along those lines to allow for the rpc style calls with crud type calls as well. 

This has pro's and con's as with anything, and could provide more tools for people to shoot themselves in the foot with, but at the same time what library already doesn't have some form of this. What are your thoughts?

**Comment by ivan-kleshnin** [comment]
We're gradually switching from REST to RPC style API in our projects. It's mostly NodeJS/Bun but some Python as well. 
We use REST frameworks to design API in RPC style: TRPC just makes everything shorter (defaulting to POST verb, for example) but it's not necessary. You can define RPC-style endpoints and generate documentation for them using all the classic BE and FE tools, it's just HTTP after all.

RPC style is so much more convenient – I keep asking myself why REST even gained its popularity 🤔

**Reply by aKardasz** [reply]
Honestly, it's not too far off. Google has the aip's that pretty much do rest with rpc styling.  At the end of the day even rpc style has the basic crud functionality in some form. So I have chilled out with the thought of it.
